qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteSnippetCompletionInIfSectionTest.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.completions;
+
+import static com.redhat.qute.QuteAssert.SECTION_SNIPPET_SIZE;
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for Qute snippet completion in #if section.
+ *
+ */
+public class QuteSnippetCompletionInIfSectionTest {
+
+	@Test
+	public void singleIf() throws Exception {
+		String template = "|";
+		testCompletionFor(template, //
+				true, // snippet support
+				c("if", //
+						"{#if ${1:condition}}\n" + //
+								"	$0\n" + //
+								"{/if}", //
+						r(0, 0, 0, 0)));
+	}
+
+	@Test
+	public void singleElse() throws Exception {
+		String template = "|";
+		testCompletionFor(template, //
+				true, // snippet support
+				c("if-else", //
+						"{#if ${1:condition}}\n" + //
+								"	$2\n" + //
+								"{#else}\n" + //
+								"	$0\n" + //
+								"{/if}", //
+						r(0, 0, 0, 0)));
+	}
+
+	@Test
+	public void singleElseIf() throws Exception {
+		String template = "|";
+		testCompletionFor(template, //
+				true, // snippet support
+				c("if-elseif", //
+						"{#if ${1:condition}}\n" + //
+								"	$2\n" + //
+								"{#else if ${3:condition}}\n" + //
+								"	$4\n" + //
+								"{#else}\n" + //
+								"	$0\n" + //
+								"{/if}", //
+						r(0, 0, 0, 0)));
+	}
+
+	@Test
+	public void snippetCompletionInNestedIf() throws Exception {
+		String template = "{#if condition}\n" + //
+				"|\n" + //
+				"{/if}";
+		testCompletionFor(template, //
+				true, // snippet support
+				SECTION_SNIPPET_SIZE /* #each, #if, ... */ + 2, /* #else #else if */ //
+				c("else", "{#else}$0", r(1, 0, 1, 0)), //

```

Comment:
Your test is correct, but just for your information, if your snippet #else will have a body with several items, it should generate lien separator. In this case it will be \n which should be used because the original template defines a \n